src/client/app/components/button.jsx

```diff
@@ -0,0 +1,17 @@
+import React from "react";
+
+class Button extends React.Component {
+
+    calculating = () => {
+        this.props.changeValue(Math.sin(this.props.value));
```

Comment:
const { value = '', changeValue } = this.props;
const sin = Math.sin(value)
changeValue(sin);